### ISSUE TITLE
Browser OS system information is inaccurate (and should be removed).

### ISSUE THREAD

**Issue by JasonElkin** [umbraco/Umbraco-CMS#13156]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

10.3.0-rc

### Bug summary

The system information section of the Backoffice help menu uses `window.Navigator.platform` which has been deprecated.

See the [note in the mdn web docs](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/platform#browser_compatibility).

### Steps to reproduce

Log into the Umbraco Backoffice on a Windows x64 machine using a modern browser. It will report "Win32".
![image](https://user-images.githubusercontent.com/7151793/194916949-d0ccf79b-8c61-4f44-b478-9f9bf74b9d25.png)

Other browsers/OS combinations will be wrong in different ways.

### Expected result / actual result

I don't actually expect the browser to know 100% what my what my operating system is. With the current APIs available it's always going to be inaccurate, so this feature should really be removed.

**Comment by JasonElkin** [comment]
Could tack this onto #13159

**Comment by nul800sebastiaan** [comment]
I would actually really, really, really like to have a best-guess operating system in there, for reasons you might understand... 

Even if it can not be quite accurate all of the time, I'd rather have information to go on when reported rather than having to go in and assume it's windows only to be told after a long discussion that we're having a linux-specific issue.

Anyway, nobody copies this info into issues anyway 😭 so it doesn't really matter at the moment what's in there. I don't know how to make it more obvious that this is super valuable information for us. 

It pastes so nicely on GitHub as well.. 

<html>
<body>


Category | Data
-- | --
Server OS | Microsoft Windows 10.0.22623
Server Framework | .NET 6.0.10
Default Language | en-US
Umbraco Version | 10.3.0-rc
Current Culture | en-US
Current UI Culture | en-US
Current Webserver | IIS
Models Builder Mode | InMemoryAuto
Debug Mode | True
Database Provider | Microsoft.Data.SqlClient
Current Server Role | Single
Browser | Firefox 105.0
Browser OS | Win32


</body>
</html>

**Comment by nul800sebastiaan** [comment]
And yes, I know I'm on Winx64, proving your point 😂 Just good to know it's windows.

**Comment by JasonElkin** [comment]
Given that use case @nul800sebastiaan, we could add a copy button there to make it easier?

We could also include the full UA string in there (in one of those friendly boxes that doesn't overflow) which will tell us the OS (or at least hint at it) most accurately.

If listing the OS explicitly is important here then we could look at using the `navigator.userAgentData` API if it's there, then fall back to looking up the OS based on the user agent (which is how this used to work).

**Comment by nul800sebastiaan** [comment]
> we could add a copy button there to make it easier?
You're making me cry..  😂 

![image](https://user-images.githubusercontent.com/304656/195370255-a7d293c1-0257-4be9-bee7-dace9ef05073.png)

> the full UA string
Please define UA? :) 

> we could look at using the `navigator.userAgentData` API if it's there, then fall back to looking up the OS based on the user agent (which is how this used to work).

Ah User Agent! Oh yes that would be nice to look at. 
I think we left the full user agent out because it was just a bit overkill? But I can't remember!

**Comment by JasonElkin** [comment]
> You're making me cry.. 😂
Oh, that's a button... maybe needs a hover state.

User Agent can be so helpful, especially with edge cases, so I think it's worth having in there - we _could_ make it so it only appears on the copy/pasted info, or just stop it looking "too much" in the UI.

I'll see if I can sensibly extend the `platformService` to use a navigator.userAgentData, with a friendly fallback.

**Comment by nikolajlauridsen** [comment]
Hey @JasonElkin just letting you know that I've had a look at this too, I agree that any OS information is better, even if it's not 100% accurate, for now I've gone ahead and marked this issue as up for grasp, should you feel so inclined 😉

**Comment by c9mb** [comment]
You should perhaps verify that you can get what's needed from that, and whether or not you need to use the new UA-Hints - given that Google are already deprecating UserAgent from Chrome. 51-degrees is the best known service for handling this requirement - either via a REST-API request or by running the service/database locally, and they do all the heavy-lifting of supporting the changing agent-strings, agent-hints and Google's intransigence - it would be a nice extension to Umbraco-Cloud.

**Comment by JasonElkin** [comment]
Ahoy @c9mb, 

Navigator.userAgentData _is_ the new UA Hints API.

The interesting thing about these new APIs is that in future it's likely browsers will ask the user to explicitly grant permission to detect the OS, so we'd need to allow for the user saying "no" and the promise failing.

If this was implemented as things stand currently, this would fire too early in the user's flow as the data is fetched as soon as the help overlay is opened - not actually when it's needed.

Browsers want to make it super hard to fingerprint, and they want to give users as much control over privacy as possible. It's not just that what we're trying to do is hard, it's that browsers are intentionally making it harder on purpose (as they should IMO).

**Comment by c9mb** [comment]
Yep, and depending on what you need, UA-Hints can make this exceptionally awkward - particularly if trying to detect things like device-type for responsive html. As I mentioned, 51-Degrees have done a lot of work on this stuff and are a good source of info - https://learnclienthints.com - obviously with a view to selling their service.

**Comment by manutdkid77** [comment]
@nul800sebastiaan Just noticed this during a routine visit to the help section and found an issue raised for it.

I was thinking of implementing a fix using the [navigator.userAgentData](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgentData) property, can I go ahead and do it ?

**Comment by nul800sebastiaan** [comment]
@manutdkid77 So sorry for the late reply! If you're still up for it, then yes, please give it a go! 

Plus it will count for [Hacktoberfest](https://umbraco.com/blog/join-us-for-umbraco-hacktoberfest-2024/) as well 👀

**Comment by manutdkid77** [comment]
No probs, ill take up this one

**Comment by madsrasmussen** [comment]
Fixed by https://github.com/umbraco/Umbraco-CMS/pull/17373
